loris/authorizer.py

```diff
@@ -257,16 +259,21 @@ def _roles_from_request(self, request):
         
         if request.path.endswith("info.json"):
             token = request.headers.get('Authorization', '')        
-            token = token.replace("Bearer", '')
+            if not isinstance(token, bytes):
+                token = token.encode('utf8')
+            token = token.replace(b"Bearer", b'')
             cval = token.strip()
             if not cval:
                 return []
-            secret = "%s-%s" % (self.token_secret, origin)
+            secret = b'-'.join([self.token_secret, origin.encode('utf8')])
```

Comment:
Rather than encoding it in multiple places, we should do the encode once, at the top of the function.